pkg/plugin/health_test.go

```diff
@@ -87,10 +137,32 @@ func TestCheckHealth(t *testing.T) {
 				}`),
 				DecryptedSecureJSONData: map[string]string{openAIKey: "abcd1234"},
 			},
-			expDetails: healthCheckResponse{
-				OpenAIEnabled: true,
-				VectorEnabled: true,
-				Version:       "unknown",
+			vService: &mockVectorService{},
+			hcClient: &mockHealthCheckClient{
+				do: func(req *http.Request) (*http.Response, error) {
+					body, _ := io.ReadAll(req.Body)
+					if strings.Contains(string(body), "gpt-4") {
+						body := io.NopCloser(strings.NewReader(`{"error": "model does not exist"}`))
+						return &http.Response{StatusCode: http.StatusNotFound, Body: body}, nil
+					}
+					return &http.Response{StatusCode: http.StatusOK, Body: io.NopCloser(strings.NewReader(""))}, nil
+				},
+			},
+			expDetails: healthCheckDetails{
+				OpenAI: openAIHealthDetails{
+					Configured: true,
+					OK:         true,
+					Error:      "",
+					Models: map[string]openAIModelHealth{
+						"gpt-3.5-turbo": {OK: true, Error: ""},
+						"gpt-4":         {OK: false, Error: `unexpected status code: 404: {"error": "model does not exist"}`},
```

Comment:
If we return the errors like `foo: bar: {error: baz}` it is hard to parse in the frontend. Could we just return an object? is this any type of convention?
I'd love to return to the frontend as much information as possible from OpenAI since this is part of the configuration phase.

This is the best we can do when the request fails when using the query:
![2023-09-29 16 08 07](https://github.com/grafana/grafana-llm-app/assets/5699976/40dd1eb6-f6a7-41ea-8f3a-766617727f41)